src/user.ts

```diff
@@ -0,0 +1,50 @@
+import { fetch } from "undici";
+import { BOT_DEVICE_ID, KEYV_BOT_STORAGE } from "./env";
+import { KeyvStorageProvider } from "./storage";
+import { LogService } from "matrix-bot-sdk";
+import { API_URL } from "./env.js";
+
+export interface UserResponse {
+	gid_uuid: string;
+	name: string;
+	display_name: string;
+	profile_photo: string;
+	created_at: string;
+	public_key: string;
+	country_code: string;
+	type: string;
+}
+
+let storage = null;
+const base_url = `${API_URL}/v1`;
+
+/**
+ * Retrieves user details from the GlobalID API or cache based on the provided GlobalID UUID.
+ * @param gid_uuid - The GlobalID UUID of the user.
+ * @returns A promise that resolves to the user details.
+ */
+export async function userDetailsResponse(gid_uuid: string) {
```

Comment:
Maybe we can rename this into `getGidIdentity`, since its a function and the name more clearly specifies its our user not matrix's?